### PR TITLE
Do not add lsp annotation when set from file

### LINKED ISSUE #562: apply/create - `local-source-proxy.apps.tanzu.vmware.com` isn't being removed when included in workload.yaml AND `--source-image` flag/value is included
### Please fill out the issue checklist below and provide ALL the requested information.

- [x] I reviewed open and closed Github issues that may be related to my problem.
- [x] I am reporting a bug that others will be able to reproduce.

### Describe the bug
apply/create - `local-source-proxy.apps.tanzu.vmware.com` isn't being removed when included in workload.yaml AND `--source-image` flag/value is included

### Expected behavior
`local-source-proxy.apps.tanzu.vmware.com` should be removed

### Steps to Reproduce
tanzu apps workload create/apply -f workload.yaml --source-image gcr.io/my-playground/lsp
```
apiVersion: carto.run/v1alpha1
kind: Workload
metadata:
  name: tanzu-java-web-app
  labels:
    apps.tanzu.vmware.com/workload-type: web
    app.kubernetes.io/part-of: tanzu-java-web-app
    apps.tanzu.vmware.com/has-tests: "true"
  annotations:
    local-source-proxy.apps.tanzu.vmware.com: my-value
spec:
  params:
    - name: annotations
      value:
        autoscaling.knative.dev/minScale: "1"
  source:
    git:
      url: https://github.com/sample-accelerators/tanzu-java-web-app
      ref:
        branch: main
```
this will result in a workload object being submitted which includes the annotation.
Update the apps plugin so the annotation is removed as expected.

### PR DESCRIPTION
**Pull request by warango4**
Pull request

### What this PR does / why we need it
Do not add LSP annotation if it comes from a file definition

### Which issue(s) this PR fixes

Fixes #562 

### Describe testing done for PR

- Added unit testing
- Created and updated workloads in local cluster

### CODE REVIEW

File: pkg/commands/workload_apply.go [hunk 1]
```diff
@@ -135,6 +135,10 @@ func (opts *WorkloadApplyOptions) Exec(ctx context.Context, c *cli.Config) error
 		workload.ReplaceMetadata(currentWorkload)
 	}
 
+	/*if fileWorkload != nil && fileWorkload.IsAnnotationExists(apis.LocalSourceProxyAnnotationName) {
```
Comment (odinnordico) [inline pkg/commands/workload_apply.go:138]: please remove the commented code

File: pkg/commands/workload_apply_test.go [hunk 2]
```diff
@@ -7751,6 +7751,119 @@ To get status: "tanzu apps workload get spring-petclinic"
 			},
 			ShouldError: true,
 		},
+		{
+			Name: "update - replace without adding lsp annotation",
```
Comment (odinnordico) [inline pkg/commands/workload_apply_test.go:7755]: is there a test using `--source-image` as mentioned in the bug?, if not can we create one

File: pkg/commands/workload_apply.go (hunk 1 above)
Comment (warango4) [reply pkg/commands/workload_apply.go]: removed

File: pkg/commands/workload_apply_test.go (hunk 2 above)
Comment (warango4) [reply pkg/commands/workload_apply_test.go:7755]: Currently, we do not have a way to test `--source-image` in the `apply` tests, I can add a e2e test and fill an issue to create tests for this kind of integration.

**Review by odinnordico — DISMISSED** [review]

**Comment by warango4** [comment]
# If workload is created from file that contains LSP annotation

Workload yaml file:

```yaml
apiVersion: carto.run/v1alpha1
kind: Workload
metadata:
  name: tanzu-web-app
  labels:
    apps.tanzu.vmware.com/workload-type: web
    app.kubernetes.io/part-of: tanzu-java-web-app
    apps.tanzu.vmware.com/has-tests: "true"
  annotations:
    local-source-proxy.apps.tanzu.vmware.com: gcr.io/tanzu-framework-playground/source:default-workload@sha256:123
spec:
  params:
    - name: annotations
      value:
        autoscaling.knative.dev/minScale: "2"
  source:
    git:
      url: https://github.com/sample-accelerators/tanzu-java-web-app
      ref:
        branch: main
```

```bash
$ tanzu apps workload apply -f workload.yaml

❗ WARNING: Configuration file update strategy is changing. By default, provided configuration files will replace rather than merge existing configuration. The change will take place in the January 2024 TAP release (use "--update-strategy" to control strategy explicitly).

🔎 Create workload:
      1 + |---
      2 + |apiVersion: carto.run/v1alpha1
      3 + |kind: Workload
      4 + |metadata:
      5 + |  labels:
      6 + |    app.kubernetes.io/part-of: tanzu-java-web-app
      7 + |    apps.tanzu.vmware.com/has-tests: "true"
      8 + |    apps.tanzu.vmware.com/workload-type: web
      9 + |  name: tanzu-web-app
     10 + |  namespace: default
     11 + |spec:
     12 + |  params:
     13 + |  - name: annotations
     14 + |    value:
     15 + |      autoscaling.knative.dev/minScale: "2"
     16 + |  source:
     17 + |    git:
     18 + |      ref:
     19 + |        branch: main
     20 + |      url: https://github.com/sample-accelerators/tanzu-java-web-app
❓ Do you want to create this workload? [yN]:
```

**Comment by warango4** [comment]
# If workload was created with LSP and it's updated from file contains LSP annotation with a different value

```yaml
apiVersion: carto.run/v1alpha1
kind: Workload
metadata:
  name: tanzu-web-app
  labels:
    apps.tanzu.vmware.com/workload-type: web
    app.kubernetes.io/part-of: tanzu-java-web-app
    apps.tanzu.vmware.com/has-tests: "true"
  annotations:
    local-source-proxy.apps.tanzu.vmware.com: gcr.io/tanzu-framework-playground/source:default-workload@sha256:123
spec:
  params:
    - name: annotations
      value:
        autoscaling.knative.dev/minScale: "2"
  source:
    git:
      url: https://github.com/sample-accelerators/tanzu-java-web-app
      ref:
        branch: main
```

```bash
tanzu apps workload apply tanzu-java-web-app --local-path . -f workload.yaml
❗ WARNING: Configuration file update strategy is changing. By default, provided configuration files will replace rather than merge existing configuration. The change will take place in the January 2024 TAP release (use "--update-strategy" to control strategy explicitly).

The files and/or directories listed in the .tanzuignore file are being excluded from the uploaded source code.
Publishing source in "." to "local-source-proxy.tap-local-source-system.svc.cluster.local/source:default-tanzu-java-web-app"...
No source code is changed

🔎 Update workload:
...
  4,  4   |metadata:
  5,  5   |  annotations:
  6,  6   |    local-source-proxy.apps.tanzu.vmware.com: gcr.io/tanzu-framework-playground/source:default-tanzu-java-web-app@sha256:447db92e289dbe3a6969521917496ff2b6b0a1d6fbff1beec3af726430ce8493
  7,  7   |  labels:
      8 + |    app.kubernetes.io/part-of: tanzu-java-web-app
      9 + |    apps.tanzu.vmware.com/has-tests: "true"
  8, 10   |    apps.tanzu.vmware.com/workload-type: web
  9, 11   |  name: tanzu-java-web-app
 10, 12   |  namespace: default
 11, 13   |spec:
     14 + |  params:
     15 + |  - name: annotations
     16 + |    value:
     17 + |      autoscaling.knative.dev/minScale: "2"
 12, 18   |  source:
 13, 19   |    image: gcr.io/tanzu-framework-playground/source:default-tanzu-java-web-app@sha256:447db92e289dbe3a6969521917496ff2b6b0a1d6fbff1beec3af726430ce8493
❓ Really update the workload "tanzu-java-web-app"? [yN]:
```

**Comment by warango4** [comment]
# If workload was created with LSP and it's being updated from a file that does not contain a source but it does contain LSP annotation using replace update strategy

```yaml
apiVersion: carto.run/v1alpha1
kind: Workload
metadata:
  name: tanzu-web-app
  labels:
    apps.tanzu.vmware.com/workload-type: web
    app.kubernetes.io/part-of: tanzu-java-web-app
    apps.tanzu.vmware.com/has-tests: "true"
  annotations:
    local-source-proxy.apps.tanzu.vmware.com: gcr.io/tanzu-framework-playground/source:default-workload@sha256:123
spec:
  params:
    - name: annotations
      value:
        autoscaling.knative.dev/minScale: "2"
```

```bash
tanzu apps workload apply tanzu-java-web-app -f ~/workload.yaml --update-strategy replace
❗ WARNING: Configuration file update strategy is changing. By default, provided configuration files will replace rather than merge existing configuration. The change will take place in the January 2024 TAP release (use "--update-strategy" to control strategy explicitly).

🔎 Update workload:
  1,  1   |---
  2,  2   |apiVersion: carto.run/v1alpha1
  3,  3   |kind: Workload
  4,  4   |metadata:
  5     - |  annotations:
  6     - |    local-source-proxy.apps.tanzu.vmware.com: gcr.io/tanzu-framework-playground/source:default-tanzu-java-web-app@sha256:447db92e289dbe3a6969521917496ff2b6b0a1d6fbff1beec3af726430ce8493
  7,  5   |  labels:
      6 + |    app.kubernetes.io/part-of: tanzu-java-web-app
      7 + |    apps.tanzu.vmware.com/has-tests: "true"
  8,  8   |    apps.tanzu.vmware.com/workload-type: web
  9,  9   |  name: tanzu-java-web-app
 10, 10   |  namespace: default
 11, 11   |spec:
 12     - |  source:
 13     - |    image: gcr.io/tanzu-framework-playground/source:default-tanzu-java-web-app@sha256:447db92e289dbe3a6969521917496ff2b6b0a1d6fbff1beec3af726430ce8493
     12 + |  params:
     13 + |  - name: annotations
     14 + |    value:
     15 + |      autoscaling.knative.dev/minScale: "2"
❗ NOTICE: no source code or image has been specified for this workload.
❓ Really update the workload "tanzu-java-web-app"? [yN]:
```

**Comment by warango4** [comment]
# If workload was created with LSP and it's updated from a file that does not contain LSP annotation

```yaml
apiVersion: carto.run/v1alpha1
kind: Workload
metadata:
  name: tanzu-web-app
  labels:
    apps.tanzu.vmware.com/workload-type: web
    app.kubernetes.io/part-of: tanzu-java-web-app
    apps.tanzu.vmware.com/has-tests: "true"
spec:
  params:
    - name: annotations
      value:
        autoscaling.knative.dev/minScale: "2"
  source:
    git:
      url: https://github.com/sample-accelerators/tanzu-java-web-app
      ref:
        branch: main
```

```bash
tanzu apps workload apply tanzu-java-web-app --local-path . -f workload.yaml
❗ WARNING: Configuration file update strategy is changing. By default, provided configuration files will replace rather than merge existing configuration. The change will take place in the January 2024 TAP release (use "--update-strategy" to control strategy explicitly).

The files and/or directories listed in the .tanzuignore file are being excluded from the uploaded source code.
Publishing source in "." to "local-source-proxy.tap-local-source-system.svc.cluster.local/source:default-tanzu-java-web-app"...
No source code is changed

🔎 Update workload:
...
  4,  4   |metadata:
  5,  5   |  annotations:
  6,  6   |    local-source-proxy.apps.tanzu.vmware.com: gcr.io/tanzu-framework-playground/source:default-tanzu-java-web-app@sha256:447db92e289dbe3a6969521917496ff2b6b0a1d6fbff1beec3af726430ce8493
  7,  7   |  labels:
      8 + |    app.kubernetes.io/part-of: tanzu-java-web-app
      9 + |    apps.tanzu.vmware.com/has-tests: "true"
  8, 10   |    apps.tanzu.vmware.com/workload-type: web
  9, 11   |  name: tanzu-java-web-app
 10, 12   |  namespace: default
 11, 13   |spec:
     14 + |  params:
     15 + |  - name: annotations
     16 + |    value:
     17 + |      autoscaling.knative.dev/minScale: "2"
 12, 18   |  source:
 13, 19   |    image: gcr.io/tanzu-framework-playground/source:default-tanzu-java-web-app@sha256:447db92e289dbe3a6969521917496ff2b6b0a1d6fbff1beec3af726430ce8493
❓ Really update the workload "tanzu-java-web-app"? [yN]:
```

**Review by shaheerkootteeri — APPROVED** [review]
LGTM

**Review by odinnordico — APPROVED** [review]

**Pull request merged**